### PR TITLE
Implement missing overloads for generic AS in generic target

### PR DESCRIPTION
**Pull request by PietroGhg**
Following up to https://github.com/intel/llvm/pull/13428, this PR implements the overloads for the generic AS that are missing in the `generic` libclc target.

### CODE REVIEW

File: libclc/generic/libspirv/atomic/atomic_cmpxchg.cl [hunk 1]
```diff
@@ -24,6 +24,13 @@ _Z29__spirv_AtomicCompareExchangePU3AS1iN5__spv5Scope4FlagENS1_19MemorySemantics
   return __sync_val_compare_and_swap(p, cmp, val);
 }
 
+_CLC_DEF int
+_Z29__spirv_AtomicCompareExchangePiN5__spv5Scope4FlagENS0_19MemorySemanticsMask4FlagES4_ii(
```
Comment (frasercrmck) [inline libclc/generic/libspirv/atomic/atomic_cmpxchg.cl:28]: Do these definitions also need to be wrapped in `_CLC_GENERIC_AS_SUPPORTED`?

Comment (PietroGhg) [reply libclc/generic/libspirv/atomic/atomic_cmpxchg.cl]: Yeah well spotted, thank you. I'd also to refactor this file to use a macro system similar to the other files, but I'd rather do that in a separate PR

**Review by frasercrmck — APPROVED** [review]
LGTM

File: libclc/generic/libspirv/atomic/atomic_cmpxchg.cl (hunk 1 above)
Comment (frasercrmck) [reply libclc/generic/libspirv/atomic/atomic_cmpxchg.cl]: Yeah, there's a lot of opportunity for that sort of thing around libclc - enough work for a good while :)

**Comment by PietroGhg** [comment]
Hi @intel/llvm-gatekeepers, this looks ready to merge, thanks

**Pull request merged**